README.md

```diff
@@ -199,15 +206,53 @@ When using resnet101 to score 50k images, the job took ~30m and cost ~1.4$ with
 For `mobilenet_v2`, it cost 0.5$ with ~22m.
 Note the cost and time depends on your job settings and the regions.
 
+#### Build Custom Container with GPU supports
 Running Dataflow GPU jobs needs to build a custom container,
 ```bash
 make docker
 ```
 The final docker image will be pushed to Artifact Registry. For this guide,
 we use `tensor_rt.Dockerfile` to demonstrate how to build the container to run the inference on GPUs with TensorRT.
 **Note given the base image issue for TensorRT, only Python 3.8 should be used when running GPUs.**
-You can follow [this doc](https://cloud.google.com/dataflow/docs/guides/using-gpus) to create other GPU containers.
+You can follow [this doc](https://cloud.google.com/dataflow/docs/gpu/use-gpus#custom-container) to create other GPU containers.
 
+#### Test Custom Container using GCE VM
+It is highly recommended to test your custom container locally before running it with Dataflow,
+```bash
+make create-vm
+```
+This creates a GCE VM with a T4 GPU and installs nvidia driver. It will take a few minutes.
+Now using this VM allows you to test whether the docker container is built correctly,
+```bash
+# check whether Beam is installed in Custom Container
+make check-beam
+# check whether Tensorflow can use GPUs in Custom Container
+make check-tf-gpu
+# check whether PyTorch can use GPUs in Custom Container
+make check-torch-gpu
+```
+Note these commands will take some time to download your container.
+You should see outputs similar to these:
+```bash
+Checking Python version on VM...
+Python 3.8.10
+Checking venv exists on VM...
+python3-venv/now 3.8.2-0ubuntu2 amd64 [installed,local]
+Checking Beam Version on VM...
+2.48.0
+Checking Tensorflow on GPU...
+[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU'), PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
+Checking PyTorch on GPU...
+True
+Tesla T4
+```
+
+After finishing tests, you can remove this VM,
```

Comment:
The docs recommend iterating on this newly created GPU using the direct runner. This will help find issues like a model not correctly loading or inference failing because of GPU specific issues, or issues with the pre/postprocessing functions.

I think we should at least drop in a sentence about this for now, but eventually it would be good to have a script that does this.

Reply:
Good point. I just added `make check-pipeline` to validate whether the pipeline with DirectRunner can run successfully with Custom Container.